### DISCUSSION TITLE
OCAPI

### DISCUSSION THREAD

**Discussion by joji-litmus7** [SalesforceCommerceCloud/sfcc-ci#448] [Team Posts]
What is the best approach to add custom attributes in "/product_search" API response?

**Comment by aventurinisalesforce** [comment]
Hi,

if the custom attribute are in the product, you could retrieve from productHit, if are some calculated attribute or external ones, is better check one to one if this process could decrease performance

**Comment by joji-litmus7** [comment]
Hi, 

Thanks for your response, but how to enable custom attributes in productHit?

**Comment by vivily** [comment]
Hi Joji,

AFAIK, as long as the attribute is showing on your BM product detail
screen, you don't need to do anything extra to enable custom attribute in
the OCAPI. If you have a selector in your request, just use wildcard or add
the custom attribute to the selector. The attribute will be in the response
document.

Wei

On Mon, Aug 5, 2019 at 6:56 AM Joji Thomas <notifications@github.com> wrote:

> Hi,
>
> Thanks for your response, but how to enable custom attributes in
> productHit?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/SalesforceCommerceCloud/teams/community/discussions/101/comments/2?email_source=notifications&email_token=AAPID6TCOS6OYHLBJKOJAUTQDABNDA5CNFSM4IJHN6X2YY3PNVWWK3TUL52HS4DFWNCGS43DOVZXG2LPNZIG643UKJSXA3DZVJRW63LNMVXHIX3JMTHAAAQXT4>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAPID6RM2PYRIAEOJVXRAALQDABNDANCNFSM4IJHN6XQ>
> .
>


-- 

<http://smart.salesforce.com/sig/wei.liu//us_mb/default/link.html>

**Comment by joji-litmus7** [comment]
Hi Vivily,

You are right, we will get all the product related attributes in GET /products/{id} API, but I need the same custom attributes in GET /product_search API response. Is there any configuration or settings for achieving this requirement.

**Comment by astound-scollins** [comment]
Hi @joji-litmus7 ,

No, there is not a configuration to do this as it really is not the best practice to add this data to that response. 

If you really want to do it, the only means by which to do so would be through the dw.ocapi.shop.product_search.modifyGETResponse hook. There you would need to get the response document and then get each of the products and add the additional fields that you want to that document. 

This means that you would be taking an API request that was just reading from an index and would be adding additional, maybe cached, maybe not, calls to get the product object for each record to get that data. 

The best practice way to get that information would be to get the array of items back from the /product_search API and then call /products/{id}....{id} to get the product records.

**Comment by saurabh-jain012** [comment]
In modifyGETResponse get the original product from getProduct api and loop over product.custom and set in c_customproperty. When you loop take care of enum values.

**Comment by jorgehernandezSF** [comment]
There is a video that covers using OCAPI hooks in the Partner Community
Learning Path:  https://embed.vidyard.com/share/SuHKKP72voLN2ezSDidqeB?
Check it out.

Jorge
